lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+                    return False
+
+        return True
+
+    @staticmethod
+    def compare_list(first_resource, second_resource):
+        """
+        Recursively compares lists contents equivalence, ignoring types and element orders.
+        Lists with same size are compared value by value after a sort,
+        each element is converted to str before the comparison.
+        Args:
+            first_resource: first list
+            second_resource: second list
+
+        Returns:
+            True when equal;
+            False when different.
+        """
+
+        resource1 = deepcopy(first_resource)
+        resource2 = deepcopy(second_resource)
+
+        debug_resources = "resource1 = {0}, resource2 = {1}".format(resource1, resource2)
+
+        # The second list is null / empty  / False
+        if not resource2:
+            logger.debug("resource 2 is null. " + debug_resources)
+            return False
+
+        if len(resource1) != len(resource2):
+            logger.debug("resources have different length. " + debug_resources)
+            return False
+
+        resource1 = sorted(resource1, key=ResourceComparator._str_sorted)
+        resource2 = sorted(resource2, key=ResourceComparator._str_sorted)
+
+        for i, val in enumerate(resource1):
+            if isinstance(val, dict):
+                # change comparison function to compare dictionaries
+                if not ResourceComparator.compare(val, resource2[i]):
+                    logger.debug("resources are different. " + debug_resources)
+                    return False
+            elif isinstance(val, list):
+                # recursive call
+                if not ResourceComparator.compare_list(val, resource2[i]):
+                    logger.debug("lists are different. " + debug_resources)
+                    return False
+            elif ResourceComparator._standardize_value(val) != ResourceComparator._standardize_value(resource2[i]):
+                logger.debug("values are different. " + debug_resources)
+                return False
+
+        # no differences found
+        return True
+
+    @staticmethod
+    def _str_sorted(obj):
+        if isinstance(obj, dict):
+            return json.dumps(obj, sort_keys=True)
+        else:
+            return str(obj)
+
+    @staticmethod
+    def _standardize_value(value):
+        """
+        Convert value to string to enhance the comparison.
+
+        Args:
+            value: Any object type.
+
+        Returns:
+            str: Converted value.
+        """
+        if isinstance(value, float) and value.is_integer():
+            # Workaround to avoid erroneous comparison between int and float
+            # Removes zero from integer floats
+            value = int(value)
+
+        return str(value)
+
+
+class ResourceMerger():
+    @staticmethod
+    def merge_list_by_key(original_list, updated_list, key, ignore_when_null=[]):
```

Comment:
If you're unfamiliar with why that is, you should probably google it.  It has to do with python processing the function declaration once when the function is declared and therefore there's only one copy of the default value which is used every time the function is called.  If you have a mutable container as a default value, it will not be recreated between invocations so it may not be empty the second time you call the function.